Maskbook/Repository/PluginStorageRepository.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  PluginStorageRepository.swift
+//  Maskbook
+//
+//  Created by Hugo L on 2022/4/18.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import CoreData
+import CoreDataStack
+import Foundation
+import SwiftyJSON
+import web3swift
+
+enum PluginStorageRepository {
+    static let viewContext = AppContext.shared.coreDataStack.persistentContainer.viewContext
+    static let backgroundContext = AppContext.shared.backgroundContext
+    
+    enum PluginID: String {
+        case redPackage = "com.maskbook.red_packet"
+    }
+    enum PluginType: String {
+        case redPackage = "red-packet"
+    }
+}
+
+// red package
+extension PluginStorageRepository {
+    struct RedPacketRecord: Codable {
+        var id: String
+        var contractVersion = "4"
+        var post: String?
+        var password: String
+        var txHash: String?
+        var type: String
+    }
+    
+    static func save(
+        address: String,
+        chain: BlockChainNetwork,
+        nonce: String,
+        record: RedPacketRecord
+    ) {
+        let key = "\(address)-\(chain.chain.rawValue)-\(chain.networkId)-\(nonce)"
```

Comment:
`"\(address)-\(chain.chain.rawValue)-\(chain.networkId)-\(nonce)"` 
This line of code appears tiwce, recommend use a function

Reply:
use `"\(address)-\(chain.rawValue)-\(nonce)"` instead.